geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStore.scala

```diff
@@ -302,16 +318,52 @@ abstract class GeoMesaDataStore[DS <: GeoMesaDataStore[DS, F, W], F <: WrappedFe
 
   /**
     * Checks that the distributed runtime jar matches the project version of this client. We cache
-    * successful checks for 10 minutes.
+    * successful checks for 24 hours.
     */
   private def checkProjectVersion(): Unit = {
-    if (projectVersionCheck.get() > System.currentTimeMillis()) {
-      projectVersionCheck.set(System.currentTimeMillis() + 3600000) // 1 hour
+    import GeoMesaDataStore.{projectVersionChecks, updateVersionCheck}
+
+    if (!suppressVersionCheck.get() &&
+        projectVersionChecks.get(getVersionCheckKey).getAndUpdate(updateVersionCheck) < System.currentTimeMillis()) {
       val (clientVersion, iteratorVersions) = getVersion
-      if (iteratorVersions.exists(_ != clientVersion)) {
-        val versionMsg = "Configured server-side iterators do not match client version - " +
-            s"client version: $clientVersion, server versions: ${iteratorVersions.mkString(", ")}"
-        logger.warn(versionMsg)
+
+      def message = "Configured server-side iterators do not match client version - " +
+          s"client version: $clientVersion, server versions: ${iteratorVersions.mkString(", ")}"
+
+      val clientMajorVersion = SemanticVersion(clientVersion).major
+      var mismatch = false
+
+      iteratorVersions.foreach { version =>
+        // use lenient parsing to account for versions like 1.3.5.1
+        val serverMajorVersion = SemanticVersion(version, lenient = true).major
+        if (serverMajorVersion != clientMajorVersion) {
```

Comment:
it may be worth having a system prop to skip this just in case there's something crazy going on...but maybe not...

Reply:
added